### PR TITLE
Accommodate wide icons in nav and admin nav

### LINKED ISSUE #40153: Wide logos cover the icon to collapse the sidebar
### Describe the bug

When using custom logos, if your logo is too big it will cover the icon to collapse the sidebar

### To Reproduce

1) add this logo
images (2)
2) see how it looks

### Expected behavior

We shouldn't cover the logo

### Logs

_No response_

### Information about your Metabase installation

```JSON
v49
```

### Severity

P2

### Additional context

P2 as you're forced to wipe the logo

### PR DESCRIPTION
**Pull request by rafpaf**
Closes #40153

### Description

Updates the main navigation bar and admin navigation bar to accommodate wide icons.

### How to verify

Describe the steps to verify that the changes are working as expected.

In Settings > Appearance (`/admin/settings/whitelabel/branding`), upload the following wide logo and confirm that the admin navigation bar and main navigation bar are still usable. Ensure that in the main navigation bar, the icon that toggles the sidebar is clearly visible. Ensure that in the admin navigation bar, the text 'Metabase Admin' is visible when the viewport is wider than 1280px.

wide-logo
Download from: https://github.com/metabase/metabase/assets/130925/7d280b32-78a4-4ac7-9070-f5aa95477d5f

Then do the same with this extremely wide logo to confirm that the nav bars remain functional regardless of how wide the logo is.

very-wide-logo
Download from: https://github.com/metabase/metabase/assets/130925/79f4c4cc-25e7-4a5a-b4e4-62cb2e0bce1f

### Demo

[Loom (2m32s)](https://www.loom.com/share/96bacb45b50c42129393feb8e555a988?sid=33e9a7ff-59c0-424b-943b-308e848d5aa3)

[Loom illustrating how the icons get scaled on narrow viewports (2m35s)](https://www.loom.com/share/67265dff349e4cc28ca62a514dfce9a2?sid=32b186bc-64be-4959-a956-935f9504f678)

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by npfitz** [comment]
Just want to verify that if the image is too large, we want to center and clip on both the left and the right, instead of just one: 
image

File: frontend/src/metabase/nav/components/AppBar/AppBarLogo.styled.tsx [hunk 1]
```diff
@@ -9,8 +9,11 @@ export const LogoLink = styled(Link)<{ isSmallAppBar: boolean }>`
   align-items: center;
   justify-content: center;
   border-radius: 0.375rem;
-  width: 2.25rem;
   height: 3.25rem;
+  min-width: 2.25rem;
+  max-width: 12rem;
+  overflow-x: hidden;
+  line-height: 0;
```
Comment (rafpaf) [inline frontend/src/metabase/nav/components/AppBar/AppBarLogo.styled.tsx:16]: Adding `line-height: 0` keeps the logo vertically aligned with the hamburger menu to its left

**Comment by rafpaf** [comment]
> Just want to verify that if the image is too large, we want to center and clip on both the left and the right, instead of just one: image

Good catch and question. A truncated logo will look bad either way, and I think the user will then realize they need to upload a narrower logo. I don't think we gain anything meaningful by just clipping on one side.

**Review by npfitz — APPROVED** [review]
aside from the truncation stuff mentioned in Slack, this should be fine 👍

**Comment by rafpaf** [comment]
This doesn't seem to work on Firefox. Investigating

**Pull request merged**